tests/tests/simple_maker_taker_fee_tests.cpp

```diff
@@ -0,0 +1,1563 @@
+         create_user_issued_asset("ICOIN", izzy, charge_market_fee, price, 3,
+                                                                izzy_market_fee_percent);
+
+         // Obtain asset object after a block is generated to obtain the final object that is commited to the database
+         generate_block();
+         const asset_object& jillcoin = get_asset("JCOIN");
+         const asset_object& izzycoin = get_asset("ICOIN");
+
+
+         //////
+         // Advance to activate hardfork
+         //////
+         generate_blocks(HARDFORK_BSIP_81_TIME);
+         generate_block();
+         trx.clear();
+         set_expiration(db, trx);
+
+
+         //////
+         // After HF, test that new values can be set
+         //////
+         // Define the new taker fees
+         uint16_t jill_maker_fee_percent = jill_market_fee_percent;
+         uint16_t jill_taker_fee_percent = jill_maker_fee_percent / 2;
+
+         uint16_t izzy_maker_fee_percent = izzy_market_fee_percent;
+         uint16_t izzy_taker_fee_percent = izzy_maker_fee_percent / 2;
+
+         // Set the new taker fee for JILLCOIN
+         asset_update_operation uop;
+         uop.issuer = jill.id;
+         uop.asset_to_update = jillcoin.get_id();
+         uop.new_options = jillcoin.options;
+         uop.new_options.extensions.value.taker_fee_percent = jill_taker_fee_percent;
+
+         trx.clear();
+         trx.operations.push_back(uop);
+         db.current_fee_schedule().set_fee(trx.operations.back());
+         sign(trx, jill_private_key);
+         PUSH_TX(db, trx); // No exception should be thrown
+
+         // Check the taker fee for JILLCOIN
+         asset_object updated_asset = jillcoin.get_id()(db);
+         uint16_t expected_taker_fee_percent = jill_taker_fee_percent;
+         BOOST_CHECK(updated_asset.options.extensions.value.taker_fee_percent.valid());
+         BOOST_CHECK_EQUAL(expected_taker_fee_percent, *updated_asset.options.extensions.value.taker_fee_percent);
+
+         // Set the new taker fee for IZZYCOIN
+         uop.issuer = izzy.id;
+         uop.asset_to_update = izzycoin.get_id();
+         uop.new_options = izzycoin.options;
+         uop.new_options.extensions.value.taker_fee_percent = izzy_taker_fee_percent;
+
+         trx.clear();
+         trx.operations.push_back(uop);
+         db.current_fee_schedule().set_fee(trx.operations.back());
+         sign(trx, izzy_private_key);
+         PUSH_TX(db, trx); // No exception should be thrown
+
+         // Check the taker fee for IZZYCOIN
+         updated_asset = izzycoin.get_id()(db);
+         expected_taker_fee_percent = izzy_taker_fee_percent;
+         BOOST_CHECK(updated_asset.options.extensions.value.taker_fee_percent.valid());
+         BOOST_CHECK_EQUAL(expected_taker_fee_percent, *updated_asset.options.extensions.value.taker_fee_percent);
+
+
+         //////
+         // After HF, create limit orders that will perfectly match
```

Comment:
Best if the test case can cover how much fees are charged when HF has passed but the asset's taker fee percentage is not set.

Reply:
That's a good suggestion @abitmore .  I added test cases on this branch which are visible at https://github.com/bitshares/bitshares-core/commits/bsip81 but for some reason they are not appearing in this PR.

Reply:
I can see the new commits. Thanks.